### PR TITLE
Add The Powder Toy port

### PR DESCRIPTION
**Pull request by circl-lastname**
This also adds ports of LuaJIT and FFTW. It often has problems with loading the online content, but that's because libcurl acts weird in general on SerenityOS.

### CODE REVIEW

File: Ports/luajit/package.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=luajit
+version=2.1.0-beta3
+depends=()
```
Comment (linusg) [inline Ports/luajit/package.sh:4]: This is not needed.

File: Ports/luajit/package.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+useconfigure=false
```
Comment (linusg) [inline Ports/luajit/package.sh:5]: This is not needed.

File: Ports/libfftw3f/package.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=libfftw3f
+version=3.3.10
+depends=()
```
Comment (linusg) [inline Ports/libfftw3f/package.sh:4]: This is not needed.

File: Ports/AvailablePorts.md [hunk 4]
```diff
@@ -198,6 +198,7 @@ Please make sure to keep this list up to date when adding and updating ports. :^
 | [`tuxracer`](tuxracer/)                | Tux Racer                                                       | 0.61                     | http://tuxracer.sourceforge.net/                                               |
 | [`tcl`](tcl/)                          | Tcl                                                             | 8.6.11                   | https://www.tcl-lang.org/                                                      |
 | [`termcap`](termcap/)                  | GNU termcap                                                     | 1.3.1                    | https://www.gnu.org/software/termutils/                                        |
+| [`thepowdertoy`](thepowdertoy/)        | The Powder Toy                                                  | 96.2.350                 | https://powdertoy.co.uk/                                                       |
```
Comment (linusg) [inline Ports/AvailablePorts.md:201]: The usual package name on Linux distros seems to be just `powdertoy`.

File: Ports/thepowdertoy/patches/0002-Open-links-and-directories-in-the-correct-program.patch [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+@@ -143,10 +143,18 @@ void OpenURI(ByteString uri)
+ 		fprintf(stderr, "cannot open URI: system(...) failed\n");
+ 	}
+ #elif defined(LIN)
++#if defined(__serenity__)
++	if (uri.BeginsWith("https:")) {
++		system(("Browser \"" + uri + "\" &; disown").c_str());
++	} else {
++		system(("FileManager \"" + uri + "\" &; disown").c_str());
++	}
```
Comment (linusg) [inline Ports/thepowdertoy/patches/0002-Open-links-and-directories-in-the-correct-program.patch:23]: We have open(1) for this, similar to the xdg-open case below.

File: Ports/thepowdertoy/patches/0001-malloc.h-doesn-t-exist-on-Serenity-but-the-code-only.patch [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+diff --git a/src/Update.cpp b/src/Update.cpp
+index 50ef666..50f4f6b 100644
+--- a/src/Update.cpp
++++ b/src/Update.cpp
+@@ -5,7 +5,7 @@
+ #ifndef WIN
+ #include <sys/param.h>
+ #endif
+-#if !defined(MACOSX) && !defined(BSD)
++#if !(defined(MACOSX) || defined(BSD) || defined(__serenity__))
```
Comment (linusg) [inline Ports/thepowdertoy/patches/0001-malloc.h-doesn-t-exist-on-Serenity-but-the-code-only.patch:20]: Let's stick to the existing style here, patches are not meant to do what _you_ think is best, but should has as few changes as possible.

**Review by linusg — CHANGES_REQUESTED** [review]

**Comment by circl-lastname** [comment]
Thank you for review

File: Ports/thepowdertoy/patches/0001-malloc.h-doesn-t-exist-on-Serenity-but-the-code-only.patch (hunk 6 above)
Comment (circl-lastname) [reply Ports/thepowdertoy/patches/0001-malloc.h-doesn-t-exist-on-Serenity-but-the-code-only.patch:20]: Oh, I thought the original style didn't work for 3 conditions

File: Ports/AvailablePorts.md [hunk 7]
```diff
@@ -198,6 +198,7 @@ Please make sure to keep this list up to date when adding and updating ports. :^
 | [`tuxracer`](tuxracer/)                | Tux Racer                                                       | 0.61                     | http://tuxracer.sourceforge.net/                                               |
 | [`tcl`](tcl/)                          | Tcl                                                             | 8.6.11                   | https://www.tcl-lang.org/                                                      |
 | [`termcap`](termcap/)                  | GNU termcap                                                     | 1.3.1                    | https://www.gnu.org/software/termutils/                                        |
+| [`powdertoy`](powdertoy/)              | The Powder Toy                                                  | 96.2.350                 | https://powdertoy.co.uk/                                                       |
```
Comment (linusg) [inline Ports/AvailablePorts.md:201]: Please keep this sorted alphabetically :)

**Comment by circl-lastname** [comment]
Oops

niedz., 20 mar 2022, 18:34 użytkownik Linus Groh ***@***.***>
napisał:

File: Ports/AvailablePorts.md (hunk 7 above)
Comment (circl-lastname) [reply Ports/AvailablePorts.md:201]: Oops

Comment (circl-lastname) [reply Ports/AvailablePorts.md]: Apparently SDLPoP is out of order too

**Pull request merged**